quantum/process_keycode/process_dynamic_macro.c

```diff
@@ -81,8 +81,7 @@ void dynamic_macro_play(keyrecord_t *macro_buffer, keyrecord_t *macro_end, int8_
     }
 
     clear_keyboard();
-
-    layer_state = saved_layer_state;
+    layer_state_set(saved_layer_state);
```

Comment:
This seems to be a better way to restore state. In particular, this fixes the weird lighting issues on my Ergodox-ez